### ISSUE TITLE
align_new_line only works on initial formatting [bug]

### ISSUE THREAD

**Issue by ButterBetzi** [MarketSquare/robotframework-tidy#507]
referring to #484.
My config (-line): 
`"SplitTooLongLine: split_on_every_arg=False: split_on_every_value=False: split_on_every_setting_arg=False: align_new_line=True"`

The new align_new_line option only works with the first time of formatting. 
If you then format again it behaves as if the option was set to false.

Example:
before:

```
*** Test Cases ***
Dummy Testcase
    [Tags]    Tag1    Tag10    Tag11    Tag12    Tag13    Tag2    Tag3    Tag4    Tag5    Tag6    Tag7    Tag8    Tag9    Tag14    Tag15
    Fail
```
after first format:
```
*** Test Cases ***
Dummy Testcase
    [Tags]    Tag1    Tag10    Tag11    Tag12    Tag13    Tag14    Tag15    Tag2    Tag3    Tag4    Tag5    Tag6
    ...       Tag7    Tag8    Tag9
    Fail
```
after second format:
```
*** Test Cases ***
Dummy Testcase
    [Tags]    Tag1    Tag10    Tag11    Tag12    Tag13    Tag14    Tag15    Tag2    Tag3    Tag4    Tag5    Tag6
    ...    Tag7    Tag8    Tag9
    Fail
```

**Comment by ButterBetzi** [comment]
Note: using 4.0.1

**Comment by bhirsz** [comment]
Ah.. I totally forgot about ``NormalizeSeparators`` transformer. It will overwrite your alignment. I'm usually running my tests for each transformer separately so I didn't notice this issue. Disabling this transformer should work as temporary workaround (but then you will not have other separators fixed).

We have several possible solutions:

1. Add ``align_new_line`` to  ``NormalizeSeparators`` and remove it from ``SplitTooLongLine``

We can move this option from ``SplitTooLongLine`` transformer to ``NormalizeSeparators``. At first run ``SplitTooLongLine`` will split the line (and use incorrect alignment, but that's fine), then ``NormalizeSeparators`` with ``align_new_line`` option will align it correctly. The benefit is that it will align split settings/keyword calls that were done manually, not only by ``SplitTooLongLine``. The downside is that the new lines would be aligned only in second run (by ``NormalizeSeparators``), and will not be aligned if someone disables ``NormalizeSeparators``.

2. Check in  ``NormalizeSeparators`` if ``SplitTooLongLine:align_new_line`` is set to True, and if so, ignore first separator for every line. The benefit is that it is easy to implement and don't require users to update their config, the downside is that lines not updated by ``SplitTooLongLine`` will not be aligned (working like it was supposed to work now)

3. Add ``align_new_line`` to  ``NormalizeSeparators`` and remain ``align_new_line`` in  ``SplitTooLongLine``
It will require configuring two transformers, but have benefit of having alignment working with both transformers - not matter if the line was split or manually updated.

What do you think? I think 3 would be best.

**Comment by ButterBetzi** [comment]
I also like solution 3 best, but option 2 also sounds fine :)

**Comment by bhirsz** [comment]
I have implemented fully implemented 3. (with tests and docs) but then I realized that it may be not the best approach. Right now ``NormalizeSeparators`` works in non-destructive way for new lines - it normalized all whitespace but leaves new lines intact. How about adding mode that will make all statements single line? 

ie from
```
[Tags]    many  tags  
...  multiline
```

to
```
[Tags]    many  tags   multiline
```

Then SplitTooLongLine would pick it up and split if necessary - and then of course align new lines. With this approach we avoid adding "alignment" to NormalizeSeparators but also resolve one of the missing features (full formatting, so ignore manual new lines -> of course only if configured). The downside is that it would format all multilines, even if you prefer to have some 
formatted in a specific way (ie 2 arguments per line). It could be disabled with # robotidy: off though.

What do you think? I have 3. option already prepared but it shouldn't be a problem to switch to this new option.

**Comment by bhirsz** [comment]
After considering this,the  update to allow to 'flatten` multilines in ``NormalizeSeparators`` can be added regardless. I will do it in the next release. You will be able to test if this configuration helps you. If not, I will make another release with implemented above third option - it's already prepared in the branch so it should be quick.

I will let you know when there is next release.

**Comment by ButterBetzi** [comment]
Ok, lets do it this way. Thank you! :)

**Comment by bhirsz** [comment]
I apologise for the delay - the change was ready but release got delayed. Mentioned change (``flatten_multilines`` for ``NormalizeSeparators``) was released with Robotidy 4.2.0 and you can test it

**Comment by ButterBetzi** [comment]
Thanks, tested it.
Sorry to tell you that but this seems to be no solution. 
If I use the new option for NormalizeSeparators I get several new problems:
1. Lines get too long (SplitTooLongLine seems to no longer have an effect)
2. the aligning to other variables e.g. in the variables section doesnt work anymore
3. A lot of lines get flattened even if I dont want them to, not only the Tag one that I need.

**Comment by bhirsz** [comment]
I will introduce the other feature then. But it is also important to fix any bugs found. Can you show me example of lines that are not split but should be?

**Comment by bhirsz** [comment]
Thanks for the configuration file. It looks fine but I will use it for tests later to see if something cames up.

**Comment by ButterBetzi** [comment]
Example code:
before:
```
*** Variables ***
${YYY} =                        aaaa
@{AAAAAAAAAAAA} =               ["bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb", "ccccccccccccccccccccccccccccccc",
...                             "ddddddddddddddddddddddddddddd", e]
@{BBBBBBBBBBB} =                [["1s", "bbbbbbbbb"], ["1s", "gggggggggggggggggg"],
...                             ["1s", "kkkkkkkkkkkkkkkkkkkkkkk"], ["1s", "ffffffffffffffffffffffff"]]
@{XXXXXXXXXXXXXXXXXXXXX} =      [2, "5s"]


*** Test Cases ***
Testcase
    # Test

    [Teardown]    Run Keywords
    ...    Keyword    aaaaaaaaaaaa    bbbbbbb    ccccccccccc    dddddddddddddd    eeeeeeeeeeeeeee    AND
    ...    Sleep    ${SLEEP}

```

after:
```
*** Variables ***
${YYY} =                        aaaa
@{AAAAAAAAAAAA} =    ["bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb", "ccccccccccccccccccccccccccccccc",
...                  "ddddddddddddddddddddddddddddd", e]
@{BBBBBBBBBBB} =    [["1s", "bbbbbbbbb"], ["1s", "gggggggggggggggggg"],
...                 ["1s", "kkkkkkkkkkkkkkkkkkkkkkk"], ["1s", "ffffffffffffffffffffffff"]]
@{XXXXXXXXXXXXXXXXXXXXX} =      [2, "5s"]


*** Test Cases ***
Testcase
    # Test

    [Teardown]    Run Keywords    Keyword    aaaaaaaaaaaa    bbbbbbb    ccccccccccc    dddddddddddddd    eeeeeeeeeeeeeee    AND    Sleep    ${SLEEP}

```

**Comment by bhirsz** [comment]
Hi, sorry for the delay, I just returned from my vacation - now I'm catching up on issues (in free time). If I recall correctly, I have feature ready to be integrated & and I need to check what to need to be fixed since last update (that didn't met expectations).

**Comment by ButterBetzi** [comment]
no need to hurry, just wanted to check if there is anything new that i missed :)

**Comment by bhirsz** [comment]
I have released 4.3.0 which contains the updated ``NormalizeSeparators`` transformer with align new lines options. Let me know if it works for you

**Comment by ButterBetzi** [comment]
Works now. Thank you!
